src/tigris.ts

```diff
@@ -16,10 +16,59 @@ import {
 	ServerMetadata
 } from "./types";
 
+import {GetAccessTokenRequest as ProtoGetAccessTokenRequest} from "./proto/server/v1/auth_pb";
+
 import {DB} from "./db";
+import {AuthClient} from "./proto/server/v1/auth_grpc_pb";
+import {Utility} from "./utility";
+
+const AuthorizationHeaderName = "authorization";
+const AuthorizationBearer = "Bearer ";
 
 export interface TigrisClientConfig {
 	serverUrl: string;
+	insecureChannel?: boolean
+	refreshToken?: string
+}
+
+class TokenSupplier {
+	private refreshToken: string;
+	private accessToken: string;
+	private nextRefreshTime: number;
+	private authClient: AuthClient;
+
+	constructor(config: TigrisClientConfig) {
+		this.authClient = new AuthClient(config.serverUrl, grpc.credentials.createSsl());
+	}
+
+	getAccessToken(): string {
+		if (this.shouldRefresh()) {
+			// refresh
+			this.authClient.getAccessToken(
+				new ProtoGetAccessTokenRequest().setRefreshToken(this.refreshToken),
+				(error, response) => {
+					if (error) {
+						// log failure
```

Comment:
this is intentionally left like this. logging will be taken care across the library in next PR.